### PR TITLE
Add code and contribution guidelines

### PR DESCRIPTION
**Pull request by kasperg**
This moves [our code guidelines](http://ting.dk/wiki/ding-code-guidelines) from ting.dk to the codebase where they can be discussed and updated by developers.

This also updates the code guidelines with the changes to the repository structure, introduction of automated analysis.

Finally contribution guidelines are added with requirements and best practices for developers when submitting pull requests.

### CODE REVIEW

**Comment by kasperg** [comment]
Through this #45 is now obsolete.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Contribution guidelines
+
+Thank you for your interest in contributing. It is greatly appreciated.
+
+# Pull requests
+
+Code contributions to this project follow this process and set of requirements:
+
+1. As a contributor you should create an issue describing the suggested change on [our Redmine installation platform.dandigbib.org](http://platform.dandigbib.org/) in Danish. This allows the business side of the project to evaluate your suggestion. If it accepted for inclusion it will be marked for review. The only exception from this rule is if the change does not affect end users in any way. This could be developer documentation changes, technical tests or changes to tools.
```
Comment (rolfmadsen) [inline CONTRIBUTING.md]: As a contributor you should create an issue describing the suggested change on http://platform.dandigbib.org/projects/ddb-cms/issues/new (Danish). To create an issue you must create an account via http://platform.dandigbib.org/account/register.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Pull requests
+
+Code contributions to this project follow this process and set of requirements:
+
+1. As a contributor you should [create an issue](http://platform.dandigbib.org/projects/ddb-cms/issues/new) describing the suggested change on [our Redmine installation platform.dandigbib.org](http://platform.dandigbib.org/projects/ddb-cms/) in Danish. This allows the business side of the project to evaluate your suggestion. If it accepted for inclusion it will be marked for review. To create an issue you must [have an account on the site](http://platform.dandigbib.org/account/register). The only exception from this rule is if the change does not affect end users in any way. This could be developer documentation changes, technical tests or changes to tools.
+2. All communication on GitHub should be in English.
+3. The title of your pull request must refer to the Redmine issue number if applicable e.g. *#[issue number]: [short summary of change]*.
+4. The description of your pull request should include a link to the issue.
+5. The body of the pull request should provide a short description of the suggested change and the reasoning behind the approach you have chosen.
```
Comment (kasperg) [inline CONTRIBUTING.md]: @xendk mentioned that using `#` as a prefix is not a good idea. Many clients will leave out lines starting with `#` as they are seen as comments.

Comment (xendk) [inline CONTRIBUTING.md]: It's standard behaviour: `git commit` will start an editor on an file with # comments. That's why it ended up as "Issue #" in the Drupal guidelines when they moved from CVS.

That being said, most commit message guidelines I've seen does not  suggest putting the issue reference at the beginning of the first line, but rather later. In most cases when a developer is looking at commit logs, the issue number is irrelevant (who can remember what bug 3187 was?), while the description is more informative. And the issue number eats up of the character limit.

Secondly, there's a difference between pull requests descriptions and commit messages.

Comment (cableman) [inline CONTRIBUTING.md]: That's right but having the issue number in the first line works with the way core is assigned issues from the business layer.

Comment (xendk) [inline CONTRIBUTING.md]: Radical suggestion: Let have the issue reference in the body of the commit, but require it in PR titles.

**Comment by cableman** [comment]
Reviewed and approved

**Pull request merged**